src/utils/proxySet.ts

```diff
@@ -0,0 +1,80 @@
+import { CUSTOM_TYPE } from '../vanilla'
+
+// properies that we don't want to expose to the end-user
+interface InternalProxySet<T> extends Set<T> {
+  data: T[]
+  // used for JSON.stringify
+  toJSON(): T[]
+  [CUSTOM_TYPE]: boolean
+}
+
+export function proxySet<T>(initialValues: Iterable<T> = []): Set<T> {
+  const set: InternalProxySet<T> = {
+    data: [...new Set(initialValues)],
+    has(value) {
+      return this.data.indexOf(value) !== -1
```

Comment:
This doesn't work if the value is a non-proxy object. `proxy(value)` solves it, but the problem is `isSupportedObject` is not exposed.

Reply:
Not sure to understand the issue here.

Reply:
```js
const obj = {}
const state = proxySet([obj])
state.has(obj) // ---> expecting true
```

Reply:
This does work (tested locally) even if `proxySet` is not wrapper by `proxy`.
![Screenshot 2022-01-11 at 13 18 23](https://user-images.githubusercontent.com/5574267/148941218-220c2023-34e3-4cc2-8fec-edc12f488f5c.png)

Reply:
What about wrapping with proxy?

Reply:
I'm a bit confused 🤔. Without proxy it returns `true` but with proxy returns `false`. Is it because I'm reading from the state instead of a snapshot. I'll have to dig a bit more
 
![Screenshot 2022-01-11 at 15 01 26](https://user-images.githubusercontent.com/5574267/148956922-fbfb9b1b-f561-4ae7-864c-e1963c17cc04.png)

Reply:
https://github.com/pmndrs/valtio/pull/318#discussion_r782055337

> ```js
> const obj = {}
> const state = proxySet([obj])
> state.has(obj) // ---> expecting true
> ```

My comment was incorrect. `state.has(obj)` should be `false`. Set.has should behave like Array.includes.